utils/messages.js

```diff
@@ -185,15 +195,54 @@ function parseMessageFile(fileName, details, type, callback) {
     }
     else {
       extractFields(
-        content, details, type, function(error, constants, fields) {
+        content, details, type, function(error, aggregate) {
           if (error) {
             callback(error);
-          }
-          else {
-            details.constants = constants;
-            details.fields    = fields;
-            details.md5       = calculateMD5(details);
-            callback(null, details);
+          } else {
+            if (type == "msg") {
+              details.constants = aggregate[0].constants;
+              details.fields    = aggregate[0].fields;
+              details.md5       = calculateMD5(details, "msg");
+              callback(null, details);
+            } else if (type == "srv") {
+              // services combine the two message types to compute the
+              // md5sum
+              var rtv = {
+                // we need to clone what's already there in details
+                // into the sub-objects
+                request: JSON.parse(JSON.stringify(details)),
+                response: JSON.parse(JSON.stringify(details))
+              };
+              rtv.request.constants = aggregate[0].constants;
+              rtv.request.fields = aggregate[0].fields;
+              rtv.response.constants = aggregate[1].constants;
+              rtv.response.fields = aggregate[1].fields;
+              rtv.request.md5 = rtv.response.md5 = calculateMD5(rtv, "srv");
```

Comment:
I had been planning to make the service md5 accessible outside the request or response objects since its the hash of both of them. So the generated service file would export
```
{
Request: ServiceRequest,
Response: ServiceResponse,
md5() { // return service md5 sum }
}
```
That seems like it might be difficult for you to duplicate here though?